packages/memory-inspector/src/browser/editable-widget/memory-editable-table-widget.tsx

```diff
@@ -0,0 +1,324 @@
+        if (value === undefined) {
+            value = dataSource[arrayOffset.toInt()]?.toString(16) ?? '';
+        }
+        return value;
+    }
+
+    protected handleClearEditClick = (): void => this.clearEdits();
+
+    protected clearEdits(address?: Long): void {
+        if (typeof address === 'number') {
+            this.pendingMemoryEdits.delete(address);
+        } else {
+            this.pendingMemoryEdits.clear();
+        }
+        this.update();
+    }
+
+    protected submitMemoryEdits = async (): Promise<void> => {
+        this.memoryEditsCompleted = new Deferred();
+        for (const edit of this.createUniqueEdits()) {
+            try {
+                await this.memoryProvider.writeMemory(edit);
+            } catch (e) {
+                console.log('Problem writing memory with arguments', edit, '\n', e);
+            }
+        }
+        this.memoryEditsCompleted.resolve();
+    };
+
+    private createUniqueEdits(): Array<DebugProtocol.WriteMemoryArguments> {
+        const addressesSubmitted = new Set<string>();
+        const edits = [];
+        for (const k of this.pendingMemoryEdits.keys()) {
+            const address = Long.fromString(k);
+            const { address: addressToSend, value: valueToSend } = this.composeByte(address, true);
+            const memoryReference = '0x' + addressToSend.toString(16);
+            if (!addressesSubmitted.has(memoryReference)) {
+                const data = Buffer.from(valueToSend, 'hex').toString('base64');
+                edits.push({ memoryReference, data });
+                addressesSubmitted.add(memoryReference);
+            }
+        }
+        return edits;
+    }
+
+    protected override getWrapperHandlers(): MemoryTable.WrapperHandlers {
+        return this.options.isFrozen
+            ? super.getWrapperHandlers()
+            : {
+                onClick: this.handleTableClick,
+                onContextMenu: this.handleTableRightClick,
+                onKeyDown: this.handleTableInput,
+                onMouseMove: this.handleTableMouseMove,
+            };
+    }
+
+    private handleTableClick = (event: React.MouseEvent): void => {
+        const target = event.target as HTMLElement;
+        if (target.classList?.contains('eight-bits')) {
+            this.highlightedField = hexStrToUnsignedLong(target.getAttribute('data-id') ?? '-0x1');
+            this.update();
+            event.stopPropagation();
+        }
+    };
+
+    protected override doHandleTableRightClick(event: React.MouseEvent): void {
+        const target = event.target as HTMLElement;
+        if (target.classList?.contains('eight-bits')) {
+            this.highlightedField = hexStrToUnsignedLong(target.getAttribute('data-id') ?? '-0x1');
+        }
+        super.doHandleTableRightClick(event);
+    }
+
+    // eslint-disable-next-line max-lines-per-function,complexity
+    private handleTableInput = (event: React.KeyboardEvent): void => {
```

Comment:
Could you please make this field protected?

Reply:
Done.